test/provisioner/provision_azure.go

```diff
@@ -8,37 +8,326 @@ package provisioner
 }
 
 func NewAzureCloudProvisioner(properties map[string]string) (CloudProvisioner, error) {
+	if err := initazureProperties(properties); err != nil {
+		return nil, err
+	}
+
 	return &AzureCloudProvisioner{}, nil
 }
 
-func (l *AzureCloudProvisioner) CreateCluster(ctx context.Context, cfg *envconf.Config) error {
-	return nil
+func (p *AzureCloudProvisioner) CreateVPC(ctx context.Context, cfg *envconf.Config) error {
+	return createResourceImpl()
 }
 
-func (l *AzureCloudProvisioner) CreateVPC(ctx context.Context, cfg *envconf.Config) error {
-	return nil
+func (p *AzureCloudProvisioner) DeleteVPC(ctx context.Context, cfg *envconf.Config) error {
+	return deleteResourceImpl()
 }
 
-func (l *AzureCloudProvisioner) DeleteCluster(ctx context.Context, cfg *envconf.Config) error {
+func (p *AzureCloudProvisioner) CreateCluster(ctx context.Context, cfg *envconf.Config) error {
+	agentPoolProperties := []containerservice.ManagedClusterAgentPoolProfile{
+		{
+			Name:               &AzureProps.NodeName,
+			Count:              to.Int32Ptr(2),
+			VMSize:             &AzureProps.InstanceSize,
+			Mode:               containerservice.AgentPoolModeSystem,
+			OsType:             containerservice.OSType(AzureProps.OsType),
+			EnableNodePublicIP: to.BoolPtr(true),
+			VnetSubnetID:       &AzureProps.SubnetID,
+		},
+	}
+
+	servicePrincipalProfile := &containerservice.ManagedClusterServicePrincipalProfile{
+		ClientID: &AzureProps.ClientID,
+		Secret:   &AzureProps.ClientSecret,
+	}
+
+	ManagedClusterProperties := &containerservice.ManagedClusterProperties{
+		ServicePrincipalProfile: servicePrincipalProfile,
+		DNSPrefix:               to.StringPtr("dnsprefix"),
+		AgentPoolProfiles: &[]containerservice.ManagedClusterAgentPoolProfile{
+			agentPoolProperties[0],
+		},
+	}
+
+	ManagedCluster := &containerservice.ManagedCluster{
+		Location:                 to.StringPtr("eastus"),
+		ManagedClusterProperties: ManagedClusterProperties,
+	}
+
+	// Create the cluster
+	_, err := AzureProps.ManagedaksClient.CreateOrUpdate(context.Background(), AzureProps.ResourceGroupName, AzureProps.ClusterName, *ManagedCluster)
+	if err != nil {
+		return err
+	}
+
+	// check if cluster is ready after creation
+	clusterReady := false
+	waitMinutes := 10
+	for i := 0; i <= waitMinutes; i++ {
+		ready, err := isAzureClusterReady(AzureProps.ResourceGroupName, AzureProps.ClusterName)
+		if err != nil {
+			continue
+		}
+		if ready {
+			clusterReady = true
+			break
+		}
+
+		time.Sleep(60 * time.Second)
+	}
+
+	if !clusterReady {
+		return fmt.Errorf("Cluster %s was created but not ready in %d minutes.\n", AzureProps.ClusterName, waitMinutes)
+	}
+
+	if err = syncKubeconfig(); err != nil {
+		return err
+	}
+
+	kubeconfigPath := kconf.ResolveKubeConfigFile()
+	if kubeconfigPath == "" {
+		fmt.Fprintln(os.Stderr, "Unabled to find a kubeconfig file")
+		return fmt.Errorf("Unabled to find a kubeconfig file")
+	}
+	// cfg = envconf.NewWithKubeConfig(kubeconfigPath)
```

Comment:
Hi @kartikjoshi21 ! I am curious why this line is comment, it didn't work to update the cfg object? That's why in `syncKubeconfig()` you had the `KUBECONFIG` environment variable updated?

Reply:
Yes it didnt work to update the cfg.
This is the reason:
```
test/provisioner/provision_azure.go:287:2: ineffectual assignment to cfg (ineffassign)
	cfg = envconf.NewWithKubeConfig(kubeconfigPath)
	^
test/provisioner/provision_azure.go:215:68: SA4009: argument cfg is overwritten before first use (staticcheck)
func (p *AzureCloudProvisioner) CreateCluster(ctx context.Context, cfg *envconf.Config) error {
                                                                   ^
```

Reply:
I will update the code later on to handle this one, Planning to submit another PR for `e2e_azure_test` related things

Reply:
Got it. Thanks for the explanation @kartikjoshi21 !